firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessionsRegistrar.kt

```diff
@@ -30,12 +32,14 @@ internal class FirebaseSessionsRegistrar : ComponentRegistrar {
     listOf(
       Component.builder(FirebaseSessions::class.java)
         .name(LIBRARY_NAME)
-        .factory { FirebaseSessions() }
+        .add(Dependency.required(FirebaseApp::class.java))
+        .factory { container -> FirebaseSessions(container.get(FirebaseApp::class.java)) }
+        .eagerInDefaultApp()
```

Comment:
I set eager because I want the first session to start at app launch. Would lazy still be good enough for that?

Reply:
I think lazy should be fine. With eager my concern is that we might impact the startup time of the application. Especially since we would be dependent on libraries like Installations and Firelog.

The only caveat here would be the start time of a session. I think the start time of a session and the application start time are different from each other. So, I think lazy should be fine.

Reply:
If I make it lazy, it never initiates on my test app if I never access the SDK. Are we sure that's what we want? When Crashlytics and Fireperf depend on it, will they always access it on every launch?